cli/src/cli_util.rs

```diff
@@ -3433,14 +3433,52 @@ fn resolve_default_command(
     Ok(string_args)
 }
 
+/// Recursively expands a single alias, detecting circular references.
+///
+/// Returns the fully expanded alias definition, with all nested aliases
+/// resolved. For example, if `b = ["log"]` and `b2 = ["b", "--no-graph"]`, then
+/// expanding `b2` returns `["log", "--no-graph"]`.
+///
+/// Returns an error if a circular reference is detected (e.g., `a = ["b"]` and
+/// `b = ["a"]`, or `a = ["a"]`).
+pub fn expand_alias_recursively(
```

Comment:
Is this more like what you meant? I've extracted the error handling as well to ensure behaviour is the same in both cases.

Reply:
I thought `resolve_alias()` can largely be reused, but it's true that we only need to expand a single argument. I'm getting less sure whether we should do something like `resolve_alias(..., vec![fake_bin_name, command_name])` or duplicate `resolve_alias()` for a single command name.